cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/submit/MasterEnvironmentSparkSubmitter.java

```diff
@@ -38,32 +45,47 @@
  */
 public class MasterEnvironmentSparkSubmitter extends AbstractSparkSubmitter {
   private final SparkExecutionService sparkExecutionService;
-  private final SparkConfig sparkConfig;
+  private final MasterEnvironment masterEnv;
+  private SparkConfig sparkConfig;
+  private List<LocalizeResource> resources;
 
   /**
    * Master environment spark submitter constructor.
    */
   public MasterEnvironmentSparkSubmitter(LocationFactory locationFactory, String hostname,
-                                         SparkRuntimeContext runtimeContext, SparkConfig sparkConfig) {
+                                         SparkRuntimeContext runtimeContext, MasterEnvironment masterEnv) {
     ProgramRunId programRunId = runtimeContext.getProgram().getId().run(runtimeContext.getRunId().getId());
     WorkflowProgramInfo workflowInfo = runtimeContext.getWorkflowInfo();
     BasicWorkflowToken workflowToken = workflowInfo == null ? null : workflowInfo.getWorkflowToken();
     this.sparkExecutionService = new SparkExecutionService(locationFactory, hostname, programRunId, workflowToken);
-    this.sparkConfig = sparkConfig;
+    this.masterEnv = masterEnv;
   }
 
   @Override
-  protected Map<String, String> getSubmitConf() {
+  protected URI getJobFile() {
+    // Master environment would always contain this file.
+    // https://github.com/cdapio/cdap/blob/develop/cdap-spark-core3_2.12/src/k8s/Dockerfile#L46
+    return URI.create("local:/opt/cdap/cdap-spark-core/cdap-spark-core.jar");
```

Comment:
it's better if this comes from the SparkConfig instead of hardcoded here.

For example, if we ever move YARN into a MasterEnvironment, this would need to be a dummy jar like the current code does.

Reply:
Ah I see.